src/library/tags/tag.h

```diff
@@ -0,0 +1,310 @@
+#pragma once
+
+#include <QJsonValue>
+#include <QVector>
+
+#include "util/assert.h"
+#include "util/macros.h"
+#include "util/math.h"
+#include "util/optional.h"
+
+namespace mixxx {
+
+namespace library {
+
+namespace tags {
+
+/// A category for tags.
+///
+/// Constraints: Lowercase ASCII, no whitespace characters
+class TagFacet final {
+  public:
+    typedef QString value_t;
+
+    static bool isValidValue(
+            const value_t& value);
+
+    /// Converts the given string into lowercase and then
+    /// removes all whitespace and non-ASCII characters.
+    static value_t clampValue(
```

Comment:
`clamp` is not a verb whose meaning is clear in the context of operating on a string. Some alternative suggestions:
  * sanitizeValue
  * removeInvalidChars
  * convertToValidChars